apix-impl/src/main/java/eu/xenit/apix/alfresco/sites/SiteService.java

```diff
@@ -43,26 +45,47 @@ public SiteService(ServiceRegistry serviceRegistry, ApixToAlfrescoConversion api
         List<SiteInfo> userSites = siteService.listSites(userId);
 
         for (SiteInfo userSite : userSites) {
-            NodeRef nodeRef = c.apix(userSite.getNodeRef());
-            String shortName = userSite.getShortName();
-            String title = userSite.getTitle();
-            String description = userSite.getDescription();
-            boolean isPublic = userSite.getIsPublic();
-            NodeRef documentLibrary = c.apix(siteService.getContainer(shortName, DOCUMENT_LIBRARY_COMPONENT));
-            NodeRef links = c.apix(siteService.getContainer(shortName, LINKS_COMPONENT));
-            NodeRef dataLists = c.apix(siteService.getContainer(shortName, DATA_LISTS_COMPONENT));
-            NodeRef wiki = c.apix(siteService.getContainer(shortName, WIKI_COMPONENT));
-            NodeRef discussions = c.apix(siteService.getContainer(shortName, DISCUSSIONS_COMPONENT));
-            Map<String, NodeRef> componentsMap = new HashMap<>();
-            componentsMap.put(DOCUMENT_LIBRARY_COMPONENT, documentLibrary);
-            componentsMap.put(LINKS_COMPONENT, links);
-            componentsMap.put(DATA_LISTS_COMPONENT, dataLists);
-            componentsMap.put(WIKI_COMPONENT, wiki);
-            componentsMap.put(DISCUSSIONS_COMPONENT, discussions);
 
-            apixSites.add(new Site(nodeRef, shortName, title, description, isPublic, componentsMap));
+                NodeRef nodeRef = c.apix(userSite.getNodeRef());
+                String shortName = userSite.getShortName();
+                String title = userSite.getTitle();
+                String description = userSite.getDescription();
+                boolean isPublic = userSite.getIsPublic();
+                Map<String, NodeRef> componentsMap;
+                // Wrapping method for NYCSANSUP-34:
+                // Alfresco-RM blocked retrieval of components due to its onw permission model
+                // this caused the entire call to fail. Since the above call `siteService.listSites(userId)` implies
+                // that the user has at least read rights on the current site, we will retrieve it anyway for display
+                // purposes. If the user does indeed not have access to the component nodes, then they should be blocked
+                // when trying to access those specifically.
+                try {
+                    componentsMap = getSiteComponents(siteService, shortName);
+                } catch (AccessDeniedException accessDeniedException) {
+                    logger.debug("User {} does not have access to a site component for site {} according to exception."
+                            + " Overriding to return a complete list of all sites this user is member of.",
+                            userId, userSite, accessDeniedException);
+                    componentsMap = AuthenticationUtil.runAsSystem(() -> {
```

Comment:
@RVanhuysseXenit I have my doubts here. If the complete list of sites is retrieved and afterwards you want to get some property from one of the sites on which you don't really have read permission, will there be another access denied exception? I haven't reproduced the issue myself. Have you?

Reply:
yes it will result in another access denied exception. If I understand the RM docs correctly, there is supposed to be a discrepancy between being able to see an item exists and being able to actually see it. For as far as I can reporduce it (eg by forcing the aspect at issue onto the documentlibrary through c4a) this seems to be intended behaviour. The test user can see the site and enter it, but they cannot retrieve info from the documentlibrary node and its children

Reply:
@RVanhuysseXenit Then I would opt for leaving out the ones on which you don't have access due to Records Management. If the list of sites the user has access to is retrieved he/she probably wants to do something with that information. In that case an access denied exception would occur, which is confusing.